babel/dates.py

```diff
@@ -1282,6 +1284,16 @@ def format_timezone(self, char, num):
                 return get_timezone_name(self.value.tzinfo, width,
                                          uncommon=True, locale=self.locale)
             return get_timezone_location(self.value.tzinfo, locale=self.locale)
+        # Included additional elif condition to add support for 'X' in timezone format
+        elif char in "xX":
+            if get_timezone_gmt(self.value, width='short', locale=self.locale)=='+0000' and char == 'X':
+                return 'Z'
+            elif num == 1:
+                return get_timezone_gmt(self.value, width='short', locale=self.locale)[:-2]
```

Comment:
[The field table](http://unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table) has good examples of all the cases.

If `width="short"` returns a value like `-0700`, it looks to be appropriate for Z/ZZ/ZZZ/X/XX/XXXX/x/xx/xxxx, so it's not wrong per se.

It looks like "GMT-7" or whatever is a localized/localizable GMT format, so that's another thing that would just need support added. I think that's outside the scope of this PR, but it's something to keep in mind.